cl/opinion_page/tests.py

```diff
@@ -210,3 +220,205 @@ def test_does_the_sitemap_have_content(self):
         # Class attributes are set, just run the test in super.
         self.expected_item_count = self.get_expected_item_count()
         super(OpinionSitemapTest, self).does_the_sitemap_have_content()
+
+
+@override_settings(
+    MEDIA_ROOT=os.path.join(settings.INSTALL_ROOT, "cl/assets/media/test/")
+)
+class UploadPublication(TestCase):
+
+    fixtures = ["tenn_test_judges.json"]
+
+    def setUp(self):
+        self.client = Client()
+        tenn_group = Group.objects.get(name="tenn_work_uploaders")
+        self.tenn_user = User.objects.create_user(
+            "learned", "learnedhand@scotus.gov", "thehandofjustice"
+        )
+        self.reg_user = User.objects.create_user(
+            "test_user", "test_user@scotus.gov", "simplepassword"
+        )
+
+        self.tenn_user.groups.add(tenn_group)
+
+        self.pdf = SimpleUploadedFile(
+            "file.pdf",
+            b"%PDF-1.trailer<</Root<</Pages<</Kids[<</MediaBox[0 0 3 3]>>]>>>>>>",
+            content_type="application/pdf",
+        )
+        self.png = SimpleUploadedFile(
+            "file.png", "file_content", content_type="image/png"
+        )
+
+        qs = Person.objects.filter(positions__court_id="tennworkcompapp")
+        self.work_comp_app_data = {
+            "case_title": "A Sample Case",
+            "lead_author": qs[0].id,
+            "second_judge": qs[1].id,
+            "third_judge": qs[2].id,
+            "docket_number": "2016-231-12332",
+            "court_str": "tennworkcompapp",
+            "pk": "tennworkcompapp",
+            "cite_volume": "2020",
+            "cite_reporter": "TN WC App.",
+            "cite_page": "1",
+            "publication_date": datetime.date(2019, 4, 13),
+        }
+
+        self.work_comp_data = {
+            "lead_author": Person.objects.filter(
+                positions__court_id="tennworkcompcl"
+            )[0].id,
+            "case_title": "A Sample Case",
+            "docket_number": "2016-231-12332",
+            "court_str": "tennworkcompcl",
+            "pk": "tennworkcompcl",
+            "cite_volume": "2020",
+            "cite_reporter": "TN WC",
+            "cite_page": "1",
+            "publication_date": datetime.date(2019, 4, 13),
+        }
+
+    def tearDown(self):
+        if os.path.exists(os.path.join(settings.MEDIA_ROOT, "pdf/2019/")):
+            shutil.rmtree(os.path.join(settings.MEDIA_ROOT, "pdf/2019/"))
+        Docket.objects.all().delete()
+        OpinionCluster.objects.all().delete()
+        Opinion.objects.all().delete()
```

Comment:
I think just doing Docket will cascade and nuke the rest, assuming they're in a docket.

Reply:
√